src/app/directives/form-connector/form-connector.directive.ts

```diff
@@ -16,16 +20,22 @@ export class FormConnectorDirective implements OnInit, OnDestroy {
   formSuccess: Subscription;
   formError: Subscription;
 
-  constructor(private formGroupDirective: FormGroupDirective) {}
+  constructor(private formGroupDirective: FormGroupDirective, private store: Store<IAppState>) {}
 
   ngOnInit(): void {
     this.formChange = this.formGroupDirective.form.valueChanges
       .pipe(
         debounceTime(this.debounce),
-        tap(value => console.log('VALUE: ', value, 'PATH: ', this.path))
+        withLatestFrom(this.store.select(getAppFormState)),
+        tap(([formValue, appForm]: [FormGroup, boolean]) => {
+          console.log('VALUE: ', formValue, 'PATH: ', this.path);
+          if (!this.formGroupDirective.form.pristine && appForm) {
+            this.store.dispatch(new SetAppFormDirty());
+          }
+        })
       )
       .subscribe();
-      console.log(this.formGroupDirective.form);
+    console.log(this.formGroupDirective.form);
```

Comment:
Why are we keeping this console log?
Is that for debug purpose?
If it is the case can we please add a TODO Add flag so we enable that with a flag.
(CVSB-15050)
I have a branch with the changes for that but I didn't get a change yet to deploy and run the tests yet however we could enable that with a flag once this PR gets in.

Reply:
Yes, it is for debugging purpose and would be removed eventually as it is used by Olu and others to debug formGroup. TODO comment is added.

Reply:
nice 👍, got a PR for this https://github.com/dvsa/cvs-app-vtm/pull/232
Trying to see with the devOps how to get this through as the config has been changed .This will require pipeline changes (very likely)